src/coreclr/tools/Common/TypeSystem/Common/FieldLayoutAlgorithm.cs

```diff
@@ -1,5 +1,6 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
+#nullable disable
```

Comment:
Would it work to just put `<Nullable>disable</Nullable>` in the System.Private.TypeLoader project?

I expect we'll need to put more of these in the type loader files as we enable more warnings. It will be an ongoing annoyance if we don't turn in off.

Reply:
That make sense, even if I prefer to have nullability checks everywhere. Can you explain why TypeSystem would not benefit form nullability annotations, except the fact that it is would be a lot of work without obvious benefits.

Reply:
Short answer: I don't like typing characters that require me to hold Shift key and `?[]?>?` makes me throw up a little.

Long answer: I don't find the mental overhead of having to reason with the C# nullable analysis productive. I rarely see a nullable-enabling pull request that can get by without having to use `!`. Before introduction of every `!` one has to spend time going over the phonebook of `[NotNullIfDoesNotReturnNull]` attributes to see if they would happen to fit.

This repo still gets `NullReferenceException` bugs in nullable enabled code because of `!`. I haven't done analysis, but it doesn't feel like the numbers of these bugs were significantly changed with nullable. In this repo we rely on code reviews for correctness in general where a null dereference is a logic error like any other and tends to be found in code reviews.

Reply:
Thanks for explanation. If a code has to be annotated with `!` I'm also against nulabiliy as is not practical if applied like this. I prefer to go with longer de-tours if nescessary (like start with applying attributes) or corrode codebase with new strict style one file at a time, until strict rules can be applied. Sometimes I even like janitorial work like this. 

I do not know how well this can work in this project. I understand that due to increased churn it may appear better in the short run, but in the long run it may be worse. I cannot advocate strongly in this position. Anyway, if there appetite for make this properly, I'm positively looking.

Reply:
> Long answer: I don't find the mental overhead of having to reason with the C# nullable analysis productive. I rarely see a nullable-enabling pull request that can get by without having to use `!`. Before introduction of every `!` one has to spend time going over the phonebook of `[NotNullIfDoesNotReturnNull]` attributes to see if they would happen to fit.

Interesting view. Hm, isn't the mental overhead of "did the code author meant this variable to be nullable or not?" much higher than seeing `!` again and again (possibly even in cases where it is not needed)?

Reply:
In additional to the obvious issues where this nullability checks should be championed across different projects, convince other maintainers and take responsibilities for all future work in that direction. Some low-level patterns in this runtime is badly expressed with nullability. TypeSystem probably wants discriminated unions as part of C#, to be able handle some conditions.

Look at
https://github.com/dotnet/runtime/pull/65137/files#diff-d72739ff6fc3330902af1563b5158cb492ba464ffe14658c28589c2999363c88R63

or 

https://github.com/dotnet/runtime/pull/65137/files#diff-b81f6b6be5bec9643a5b20c37624e03d228599247c466f5d1fd3db47de2b2584R70